sklearn/linear_model/ridge.py

```diff
@@ -367,11 +367,29 @@ def _ridge_regression(X, y, alpha, sample_weight=None, solver='auto',
                       return_n_iter=False, return_intercept=False,
                       X_scale=None, X_offset=None):
 
-    if return_intercept and sparse.issparse(X) and solver != 'sag':
-        if solver != 'auto':
-            warnings.warn("In Ridge, only 'sag' solver can currently fit the "
-                          "intercept when X is sparse. Solver has been "
-                          "automatically changed into 'sag'.")
+    has_sw = sample_weight is not None
```

Comment:
Now that this test is does only once, I think that it would be more readable to inline it.

Reply:
Sorry, stupid comment: it is used inside a closure.

Reply:
Actually, I would prefer the test to be inlined, and the function too.